### ISSUE TITLE
[Bug]: ActivityNotFoundException: No Activity found to handle Intent { act=android.intent.action.VIEW dat=appboy://close/...

### ISSUE THREAD

**Issue by chenxiangcxc** [braze-inc/braze-android-sdk#30]
### Braze Android SDK Version

30.2.0

### Steps To Reproduce

Not sure, but we observe the crash from Firebase.

### Expected Behavior

No crash

### Actual Incorrect Behavior

ActivityNotFoundException is not handled, which caused app crash.

### Verbose Logs

```shell
Fatal Exception: org.chromium.base.JniAndroid$UncaughtExceptionException: Native stack trace:
#00 pc 0x00000000040ef587 /data/app/~~7bjPe4Gt3i2zigI4H0azzw==/com.google.android.trichromelibrary_626111933-LYkp38bkj02Depk3cUgkoQ==/base.apk (offset 0x8dc000)
#01 pc 0x0000000002c9a367 /data/app/~~7bjPe4Gt3i2zigI4H0azzw==/com.google.android.trichromelibrary_626111933-LYkp38bkj02Depk3cUgkoQ==/base.apk (offset 0x8dc000)
#02 pc 0x0000000002c9926f /data/app/~~7bjPe4Gt3i2zigI4H0azzw==/com.google.android.trichromelibrary_626111933-LYkp38bkj02Depk3cUgkoQ==/base.apk (offset 0x8dc000)
#03 pc 0x000000000472cadb /data/app/~~7bjPe4Gt3i2zigI4H0azzw==/com.google.android.trichromelibrary_626111933-LYkp38bkj02Depk3cUgkoQ==/base.apk (offset 0x8dc000)
#04 pc 0x00000000056f7163 /data/app/~~7bjPe4Gt3i2zigI4H0azzw==/com.google.android.trichromelibrary_626111933-LYkp38bkj02Depk3cUgkoQ==/base.apk (offset 0x8dc000)
#05 pc 0x000000000564d0fb /data/app/~~7bjPe4Gt3i2zigI4H0azzw==/com.google.android.trichromelibrary_626111933-LYkp38bkj02Depk3cUgkoQ==/base.apk (offset 0x8dc000)
#06 pc 0x00000000051f9877 /data/app/~~7bjPe4Gt3i2zigI4H0azzw==/com.google.android.trichromelibrary_626111933-LYkp38bkj02Depk3cUgkoQ==/base.apk (offset 0x8dc000)
#07 pc 0x00000000066ce8a7 /data/app/~~7bjPe4Gt3i2zigI4H0azzw==/com.google.android.trichromelibrary_626111933-LYkp38bkj02Depk3cUgkoQ==/base.apk (offset 0x8dc000)
#08 pc 0x0000000002c0613b /data/app/~~7bjPe4Gt3i2zigI4H0azzw==/com.google.android.trichromelibrary_626111933-LYkp38bkj02Depk3cUgkoQ==/base.apk (offset 0x8dc000)
#09 pc 0x0000000002c061d7 /data/app/~~7bjPe4Gt3i2zigI4H0azzw==/com.google.android.trichromelibrary_626111933-LYkp38bkj02Depk3cUgkoQ==/base.apk (offset 0x8dc000)
#10 pc 0x00000000039c5113 /data/app/~~7bjPe4Gt3i2zigI4H0azzw==/com.google.android.trichromelibrary_626111933-LYkp38bkj02Depk3cUgkoQ==/base.apk (offset 0x8dc000)
#11 pc 0x00000000039c401b /data/app/~~7bjPe4Gt3i2zigI4H0azzw==/com.google.android.trichromelibrary_626111933-LYkp38bkj02Depk3cUgkoQ==/base.apk (offset 0x8dc000)
#12 pc 0x0000000002c26323 /data/app/~~7bjPe4Gt3i2zigI4H0azzw==/com.google.android.trichromelibra<truncated: 553 chars>
       at org.chromium.base.JniAndroid.handleException(chromium-TrichromeWebViewGoogle6432.aab-stable-626111933:11)
       at android.os.MessageQueue.nativePollOnce(MessageQueue.java)
       at android.os.MessageQueue.next(MessageQueue.java:335)
       at android.os.Looper.loopOnce(Looper.java:187)
       at android.os.Looper.loop(Looper.java:319)
       at android.app.ActivityThread.main(ActivityThread.java:8913)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:608)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1103)


Caused by android.content.ActivityNotFoundException: No Activity found to handle Intent { act=android.intent.action.VIEW dat=appboy://close/... }
       at android.app.Instrumentation.checkStartActivityResult(Instrumentation.java:2252)
       at android.app.Instrumentation.execStartActivity(Instrumentation.java:1878)
       at android.app.Activity.startActivityForResult(Activity.java:5743)
       at androidx.activity.ComponentActivity.startActivityForResult(ComponentActivity.java:754)
       at android.app.Activity.startActivityForResult(Activity.java:5701)
       at androidx.activity.ComponentActivity.startActivityForResult(ComponentActivity.java:735)
       at android.app.Activity.startActivity(Activity.java:6199)
       at android.app.Activity.startActivity(Activity.java:6166)
       at com.braze.ui.inappmessage.views.InAppMessageHtmlBaseView$messageWebView$7.onCreateWindow(InAppMessageHtmlBaseView.kt:150)
       at WV.z8.addNewContents(chromium-TrichromeWebViewGoogle6432.aab-stable-626111933:39)
       at android.os.MessageQueue.nativePollOnce(MessageQueue.java)
       at android.os.MessageQueue.next(MessageQueue.java:335)
       at android.os.Looper.loopOnce(Looper.java:187)
       at android.os.Looper.loop(Looper.java:319)
       at android.app.ActivityThread.main(ActivityThread.java:8913)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:608)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1103)
```

**Comment by bryanlogan** [comment]
Are you doing any customization of `IInAppMessageWebViewClientListener`?

Also, if you have any idea what campaign this might be, that could help a lot in trying to recreate. It might be better to contact Braze support with that information rather than posting that here. You can reference this GitHub issue when you contact them.

**Comment by chenxiangcxc** [comment]
Hey @bryanlogan Thanks for the timely reply. 
1. We don't have any customization on `IInAppMessageWebViewClientListener`.
2. Unfortunately we can't tell more how to reproduce the issue on customer side, but from the crash log, the issue is clear:

The crash occurs at: 
`com.braze.ui.inappmessage.views.InAppMessageHtmlBaseView$messageWebView$7.onCreateWindow(InAppMessageHtmlBaseView.kt:150)`
<img width="732" alt="image" src="https://github.com/braze-inc/braze-android-sdk/assets/8325272/2a9512b8-60cd-45ee-8faa-a56764ea4973">

Could we simply add a `try...catch...` around `context.startActivity(browserIntent)`? I believe the issue could be fix then.

**Comment by bryanlogan** [comment]
The try/catch will mask the issue, but not fix it. I want to try and make sure something else isn't going on so your clients don't get a message, click on it, and nothing happens.

So we'll definitely put a try/catch in there, but if you could email support with your campaign, I'd like to try some tests on that.

I'm assuming you've tested this yourself and it wouldn't recreate and all worked properly, right?

**Comment by chenxiangcxc** [comment]
Hi @bryanlogan,

We don't have a way to test in our app as the code is inside Braze SDK.

For the deep reason, maybe we could check whether it is a valid case that opening uri "appboy://close/.." in browser by the code?
```
val browserIntent = Intent(Intent.ACTION_VIEW, Uri.parse(data))
context.startActivity(browserIntent)

```

More info:
1. We have 6 cases only so far, one common they are all using Android 14.
2. We are using Braze sdk `com.braze:android-sdk-ui:30.2.0` now with this issue, previously we were using `com.appboy:android-sdk-ui:24.1.0` and the old sdk looked good without this crash.

**Comment by bryanlogan** [comment]
OK. So playing around with it, if I make an HTML InApp Message with  `<A HREF="appboy://close">Close</A>` things work fine. This gets handled in https://github.com/braze-inc/braze-android-sdk/blob/2591eb63f590742983401e3f8a666d5cacd10c4b/android-sdk-ui/src/main/java/com/braze/ui/inappmessage/utils/InAppMessageWebViewClient.kt#L172

But if I use `<A HREF="appboy://close" target="_blank">Close</A>`, the click is attempted to open externally, so it won't go to the WebViewClient, and attempts to open up via an intent.

Can you check your campaign to see if you're trying to open the "appboy://close" in an external window?  Another alternative to close is `<a href="#" onclick="appboyBridge.closeMessage()">X</a>`

Once I have further details, then I'll know if we need just the try/catch or something more.

Thanks.

**Comment by radixdev** [comment]
Hi @chenxiangcxc ,

This was addressed in version 31.0.0 of the SDK. Please feel free to reopen this issue if you have followups or more questions. Thank you for filing!
